lib/mapping/object-selector.js

```diff
@@ -133,17 +135,23 @@ class ObjectSelector {
               throw new Error(`Table "${info.tables[i].name}" could not be retrieved`);
             }
 
-            // All partition and clustering keys from the table should be included in the document
-            const keyNames = table.partitionKeys.concat(table.clusteringKeys).map(k => k.name);
-            const columns = propertiesInfo.map(m => m.columnName);
+            if (keysAreIncluded(table.partitionKeys, propertiesInfo) !== keyMatches.all) {
+              // Not all the partition keys are covered
+              return false;
+            }
 
-            for (let i = 0; i < keyNames.length; i++) {
-              if (columns.indexOf(keyNames[i]) === -1) {
-                return false;
-              }
+            const clusteringKeyMatches = keysAreIncluded(table.clusteringKeys, propertiesInfo);
+
+            // All clustering keys should be included or it can be inserting a static column value
+            if (clusteringKeyMatches === keyMatches.all) {
+              return true;
+            }
+
+            if (clusteringKeyMatches === keyMatches.some) {
+              return false;
             }
 
-            return true;
+            return propertiesInfo.length === table.partitionKeys.length + 1 && hasStaticColumn(table);
```

Comment:
I believe this won't work if you wanted to update multiple static columns:

```
Error: No table matches (all PKs have to be specified) fields: [id1,s0,s1]
```

I think to fix it, we can account the number of static columns and add that to the expected count.

Reply:
Here's a patch that should fix it:

```patch
diff --git a/lib/mapping/object-selector.js b/lib/mapping/object-selector.js
index 57f956c..4d51397 100644
--- a/lib/mapping/object-selector.js
+++ b/lib/mapping/object-selector.js
@@ -151,7 +151,8 @@ class ObjectSelector {
               return false;
             }
 
-            return propertiesInfo.length === table.partitionKeys.length + 1 && hasStaticColumn(table);
+            const staticColumns = staticColumnCount(table);
+            return propertiesInfo.length === table.partitionKeys.length + staticColumns && staticColumns > 0;
           });
 
         if (filteredTables.length === 0) {
@@ -298,13 +299,11 @@ function keysAreIncluded(keys, propertiesInfo) {
 }
 
 function hasStaticColumn(table) {
-  for (let i = 0; i < table.columns.length; i++) {
-    if (table.columns[i].isStatic) {
-      return true;
-    }
-  }
+  return staticColumnCount(table) > 0;
+}
 
-  return false;
+function staticColumnCount(table) {
+  return table.columns.reduce((acc, column) => acc + (column.isStatic ? 1 : 0), 0);
 }
 
-module.exports = ObjectSelector;
\ No newline at end of file
+module.exports = ObjectSelector;
diff --git a/test/integration/short/mapping/mapper-test-helper.js b/test/integration/short/mapping/mapper-test-helper.js
index 503db0b..2dee724 100644
--- a/test/integration/short/mapping/mapper-test-helper.js
+++ b/test/integration/short/mapping/mapper-test-helper.js
@@ -57,6 +57,7 @@ const mapperHelper = module.exports = {
       PRIMARY KEY (userid))`,
       `CREATE TABLE video_rating (videoid uuid, rating_counter counter, rating_total counter, PRIMARY KEY (videoid))`,
       `CREATE TABLE table_static1 (id1 text, id2 text, s text static, value text, PRIMARY KEY (id1, id2))`,
+      `CREATE TABLE table_static2 (id1 text, id2 text, s0 text static, s1 text static, value text, PRIMARY KEY (id1, id2))`,
 
       // Insert test data
       `INSERT INTO videos (videoid, name, userid, description, location, location_type, preview_thumbnails, tags,
@@ -165,7 +166,8 @@ const mapperHelper = module.exports = {
           },
           mappings: new UnderscoreCqlToCamelCaseMappings()
         },
-        'Static': { tables: ['table_static1'] }
+        'Static': { tables: ['table_static1'] },
+        'Static2': { tables: ['table_static2'] }
       }
     });
   },
@@ -181,4 +183,4 @@ const mapperHelper = module.exports = {
       assert.deepEqual(row[columnName], doc[mapperHelper.getPropertyName(columnName)]);
     });
   }
-};
\ No newline at end of file
+};
diff --git a/test/integration/short/mapping/model-mapper-tests.js b/test/integration/short/mapping/model-mapper-tests.js
index fbb15dc..7f83c93 100644
--- a/test/integration/short/mapping/model-mapper-tests.js
+++ b/test/integration/short/mapping/model-mapper-tests.js
@@ -36,6 +36,7 @@ describe('ModelMapper', function () {
   const videoMapper = mapper.forModel('Video');
   const userMapper = mapper.forModel('User');
   const staticMapper = mapper.forModel('Static');
+  const static2Mapper = mapper.forModel('Static2');
 
   describe('#find()', () => {
     it('should use the correct table', () => {
@@ -317,6 +318,16 @@ describe('ModelMapper', function () {
           client.execute('SELECT id1, id2, s FROM table_static1 WHERE id1 = ?', [ doc.id1 ], { prepare: true }))
         .then(rs => assertRowMatchesDoc(rs.first(), doc));
     });
+
+    it('should support inserting multiple static column values without clustering keys', () => {
+
+      const doc = { id1: Uuid.random().toString(), s0: 'static value 1', s1: 'static value 2' };
+
+      return static2Mapper.insert(doc)
+        .then(() => client.execute('SELECT id1, s0, s1 FROM table_static2 WHERE id1 = ?', [ doc.id1 ], { prepare: true }))
+        .then(rs => assertRowMatchesDoc(rs.first(), doc));
+    });
+
   });
 
   describe('#update()', () => {
```

Reply:
Nice catch and fix!